drivers/serial/Kconfig.realtek_rts5912

```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2024 Realtek Semiconductor Corporation, SIBG-SD7
+#
+
+config UART_RTS5912
+	bool "UART driver for Realtek RTS5912 EC"
+	default y
+	depends on DT_HAS_REALTEK_RTS5912_UART_ENABLED
+	select UART_NS16550
+	select UART_INTERRUPT_DRIVEN
+	select PINCTRL
+	select CLOCK_CONTROL
```

Comment:
This will fail subtly. Select should be used only for helper symbols.

Details: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#select-pitfalls

Reply:
We knew your concern of the dependency and moved the configurations of `UART_NS16550` and `UART_INTERRUPT_DRIVEN` to the `defconfig` under the `boards` directory.